pkg/storagev2/chunkstore.go

```diff
@@ -17,21 +17,42 @@ type Getter interface {
 	// Get a chunk by its swarm.Address. Returns the chunk associated with
 	// the address alongside with its postage stamp, or a storage.ErrNotFound
 	// if the chunk is not found.
+	// If the chunk has multiple stamps, then the first stamp is returned in this
+	// query. In order to deterministically get the stamp, use the GetterWithStamp
+	// interface.
 	Get(context.Context, swarm.Address) (swarm.Chunk, error)
 }
 
+// GetterWithStamp is the interface that provides a way to specify a postage
+// stamp to filter the Get query. This is required as we will support multiple stamps
+// on a chunk. As a result some components would need to query chunk and provide
+// the stamp to return by using the argument.
+type GetterWithStamp interface {
+	GetWithStamp(context.Context, swarm.Address, []byte) (swarm.Chunk, error)
```

Comment:
the name should technically be `GetWithBatch` as the ID is batchID, not stamp

Reply:
So the idea behind the name is that it returns swarm.Chunk with the Stamp accessor filled using the batch ID. Hence the name WithStamp. Also, I think batch is a more generic term so it would be better to call it stamp as the overall concept. Wdyt?